src/rules/selector-attribute-square-brackets-space-inside/index.js

```diff
@@ -0,0 +1,81 @@
+import selectorParser from "postcss-selector-parser"
+import {
+  report,
+  ruleMessages,
+  styleSearch,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "selector-attribute-square-brackets-space-inside"
+
+export const messages = ruleMessages(ruleName, {
+  expectedOpening: "Expected single space after \"[\"",
+  rejectedOpening: "Unexpected whitespace after \"[\"",
+  expectedClosing: "Expected single space before \"]\"",
+  rejectedClosing: "Unexpected whitespace before \"]\"",
+})
+
+export default function (expectation) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [
+        "always",
+        "never",
+      ],
+    })
+    if (!validOptions) { return }
+
+    root.walkRules(rule => {
+      selectorParser(selectorTree => {
+        selectorTree.eachAttribute(attributeNode => {
+          const attributeSelectorString = attributeNode.toString()
+
+          styleSearch({ source: attributeSelectorString, target: "[" }, match => {
+            const nextCharIsSpace = attributeSelectorString[match.startIndex + 1] === " "
+            if (nextCharIsSpace && expectation === "never") {
+              report({
```

Comment:
Can we minimise the duplication within `report()` by using our idiomatic `complain()` pattern as [seen here](https://github.com/stylelint/stylelint/blob/master/src/rules/comment-whitespace-inside/index.js#L45-L66)?